### PR TITLE
[WIP] Apply automatic board list formatting from #10111 to Makefiles

### PR DESCRIPTION
**Pull request by x3ro**
### Contribution description

Applies the formatting mechanism from #10111 to all relevant Makefiles. Still marked as WIP since there are some corner cases to be fixed.

### CODE REVIEW

**Comment by kaspar030** [comment]
am I the only one who's bothered by having to scroll down first thing after opening a test's Makefile?

File: tests/socket_zep/Makefile [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
 
-BOARD_WHITELIST = native    # socket_zep is only available on native
+BOARD_WHITELIST = # \
+                  available \
+                  is \
+                  native \
+                  native \
+                  on \
+                  only \
+                  socket_zep
```
Comment (gebart) [inline tests/socket_zep/Makefile]: Oops!

**Comment by gebart** [comment]
@kaspar030 
> am I the only one who's bothered by having to scroll down first thing after opening a test's Makefile?

Not sure what you mean? Do you mean that the board blacklist/whitelists are too long and you have to scroll down to see the USEMODULE etc lines in the Makefile?

**Comment by kaspar030** [comment]
> Do you mean that the board blacklist/whitelists are too long and you have to scroll down to see the USEMODULE etc lines in the Makefile?

Yes...

File: tests/socket_zep/Makefile (hunk 1 above)
Comment (x3ro) [reply tests/socket_zep/Makefile]: See #10111

**Comment by x3ro** [comment]
> am I the only one who's bothered by having to scroll down first thing after opening a test's Makefile?

So far yes

**Comment by waehlisch** [comment]
You can just search: ```/USEMODULE```

**Comment by kaspar030** [comment]
> You can just search: `/USEMODULE`

Yup. Everytime I open the file, which might be hundred times more often than I modify the lists. Multiplied by the number of people that read at our code, that is quite often.

To me it seems like the board lists get much more screen real estate as they are relevant.
See [this](https://github.com/x3ro/RIOT/blob/pr/consistently-format-board-lists/examples/dtls-echo/Makefile) vs [this](https://github.com/RIOT-OS/RIOT/blob/master/examples/dtls-echo/Makefile).

If there's a script that does the formatting anyways, why not have that output lines that fill 80 characters?

**Comment by MrKevinWeiss** [comment]
Maybe we just move the BOARD_* below the USEMODULE?

**Comment by kaspar030** [comment]
> Maybe we just move the BOARD_* below the USEMODULE?

Yeah, why not? as far down as possible.

**Comment by x3ro** [comment]
> If there's a script that does the formatting anyways, why not have that output lines that fill 80 characters?

To make diffs clean, relevant and reviewable.

> Maybe we just move the BOARD_* below the USEMODULE?

Seems doable

**Comment by A-Paul** [comment]
What about including from (a) separate files(s)?

**Comment by x3ro** [comment]
> What about including from (a) separate files(s)?

Could you elaborate how this would improve the situation over putting the lists at the end of the file?

File: tests/mpu_stack_guard/Makefile [hunk 2]
```diff
@@ -1,48 +1,196 @@
...
 
 USEMODULE += mpu_stack_guard
 
-include $(RIOTBASE)/Makefile.include
-
 ifeq (llvm,$(TOOLCHAIN))
   CFLAGS += -Wno-infinite-recursion
 endif
+
+BOARD_WHITELIST := \
```
Comment (MrKevinWeiss) [inline tests/mpu_stack_guard/Makefile]: Maybe manually alter this so it doesn't look so silly?  I don't it is worth the effort adding this corner case into the script.

File: tests/socket_zep/Makefile [hunk 3]
```diff
@@ -12,4 +10,14 @@ CFLAGS += -DDEVELHELP
 
 TERMFLAGS ?= -z [::1]:17754
 
+BOARD_WHITELIST := \
```
Comment (MrKevinWeiss) [inline tests/socket_zep/Makefile]: Also manually fix this guy.

**Comment by MrKevinWeiss** [comment]
Darn...

**Comment by MrKevinWeiss** [comment]
Maybe see if the updated script works on this.  If there are only a few corner cases that go against convention I think manual fixes are fine.

**Comment by x3ro** [comment]
Closing this. I'd prefer to open a new PR with a clean discussion.

**Pull request closed without merging**